gen/installer/bash.py

```diff
@@ -261,6 +261,10 @@
     # error exit codes
     set +e
     echo -e "${BOLD}Running preflight checks${NORMAL}"
+    AGENT_ONLY=0
+    if [[ $ROLES = "slave" || $ROLES = "slave_public" ]]; then
```

Comment:
Roles is technically an array currently, so we need to iterate through all the roles to check (Moving it to not be an array because really we only ever want one role is in #303, still WIP).